.github/workflows/build.yml

```diff
@@ -48,7 +48,19 @@ jobs:
                 if: matrix.stability == 'dev'
                 run: composer config minimum-stability ${{ matrix.stability }}
 
-            -   name: Install dependencies
+            -   name: Locate composer cache
+                id: composercache
+                run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            -   name: Cache composer dependencies
+                uses: actions/cache@v2
+                env:
+                    cache-name: cache-composer
+                with:
+                    path: ${{ steps.composercache.outputs.dir }}
+                    key: ${{ hashFiles('composer.json') }}
```

Comment:
Should we include the PHP version in the cache key?

Reply:
Nice catch, if we're using multiple PHP versions to tests, then this is strongly recommended!

Reply:
here, we're caching the composer files cache, not the vendor folder. So it might be fine to share that cache between PHP versions.

@ciaranmcnulty have you compared the time it takes with Composer 2 ? This would make the job download 1 big archive for the persistent cache, vs downloading several small archives in parallel (with similar network latencies as the packages also come from github servers). the parallelisation of downloads might make this caching useless on the CI server (the cache for packagist metadata is probably useless already as it will likely be outdated for next builds, and they are small files now)

Reply:
> here, we're caching the composer files cache, not the vendor folder. So it might be fine to share that cache between PHP versions.

True as well!

Reply:
Yes I guess the cache folder can store multiple versions of the resolved dependencies. It makes me wonder whether we really do want to invalidate it when composer.json changes?

Reply:
Won't parallel runs overwrite each other's caches?

Reply:
No idea! Good question though

Reply:
Looking at the last build, composer re-downloaded 7.2 dependencies. For other PHP versions dependencies were extracted from cache.